onset_maproom/maproom.py

```diff
@@ -91,6 +98,92 @@ def initialize(toto):
     return center_of_the_map, lat_min, lat_max, lon_min, lon_max, lat_label, lon_label
 
 
+def adm_borders(shapes):
+    dbpool = DBPOOL
+    with dbpool.take() as cm:
+        conn = cm.resource
+        with conn:  # transaction
+            s = sql.Composed(
+                [
+                    sql.SQL("with g as ("),
+                    sql.SQL(shapes),
+                    sql.SQL(
+                        """
+                        )
+                        select
+                            g.label, g.key, g.the_geom
+                        from g
+                        """
+                    ),
+                ]
+            )
+            df = pd.read_sql(
+                s,
+                conn,
+            )
+    df["the_geom"] = df["the_geom"].apply(lambda x: wkb.loads(x.tobytes()))
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    return {"features": shapes}
```

Comment:
This function retrieves shapes from postgis in wkb form, and then uses shapely to convert from wkb to geojson, which is what leaflet expects. Shapely seems unnecessary here, because postgis can generate geojson directly. Try replacing ST_AsBinary with ST_AsGeoJSON in the query (which is in the config file) and replacing lines 124-129 with
```
return {"features": df["the_geom"]}
```

Reply:
I did what you said but the geometries just don't show, without error information. Well there are warnings but those have been around since day 1 when I picked this code from fbfmaproom, and they are not trivial for me to interpret. I also tried to apply ST_Transform(the_geom, 4326) but that didn't make a difference.
I printed what is returned in both set ups and the prints looked very similar, except with some syntax variations that seemed harmless, but also parentheses turned brackets...

Reply:
This is now an Issue